core/src/main/java/org/elasticsearch/index/query/MoreLikeThisQueryBuilder.java

```diff
@@ -944,23 +938,12 @@ private static void handleExclude(BooleanQuery boolQuery, List<Item> likeItems)
     }
 
     @Override
-    public QueryValidationException validate() {
-        QueryValidationException validationException = null;
-        if (likeTexts.isEmpty() && likeItems.isEmpty()) {
-            validationException = addValidationError("requires 'like' to be specified.", validationException);
-        }
-        if (fields != null && fields.isEmpty()) {
-            validationException = addValidationError("requires 'fields' to be specified", validationException);
-        }
-        return validationException;
-    }
-
-    @Override
     protected MoreLikeThisQueryBuilder doReadFrom(StreamInput in) throws IOException {
-        MoreLikeThisQueryBuilder moreLikeThisQueryBuilder = new MoreLikeThisQueryBuilder((List<String>) in.readGenericValue());
-        moreLikeThisQueryBuilder.likeTexts = (List<String>) in.readGenericValue();
+        List<String> fields = (List<String>) in.readGenericValue();
+        List<String> likeTexts = (List<String>) in.readGenericValue();
```

Comment:
`fields` can be `null` in which case the field is resolved to the default field later, so writing them as array would require extra null-handling which write/readGeneric already provides. For `likeTexts` this would be possible, but is it worth the extra toArray()/copy action on both read/write? Or do you suggest moving to internal use of `String[]` for `fields` and `likeTexts`?